src/api/java/com/minecolonies/api/colony/IColony.java

```diff
@@ -452,4 +455,26 @@ default List<BlockPos> getWayPoints(@NotNull BlockPos position, @NotNull BlockPo
      * @param component the message.
      */
     void notifyColonyManagers(Component component);
+
+    /**
+     * Adds a position on which a gatemarker is placed.
+     *
+     * @param pos The position of the gatemarker.
+     * @param gateState The state the gate is supposed to be
+     */
+    void addGateMarker(BlockPos pos,  final BlockState gateState);
```

Comment:
I feel like those should go inside the traveling manager, or into the registered structure manager

Reply:
Will make a `POIManager` for both waypoints and gates.